xla/service/gpu/openxla/default/compiler.h

```diff
@@ -0,0 +1,33 @@
+/* Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef XLA_SERVICE_GPU_OPENXLA_DEFAULT_COMPILER_H_
+#define XLA_SERVICE_GPU_OPENXLA_DEFAULT_COMPILER_H_
+
+#include <cstdlib>
+#include <string>
+
+namespace xla::gpu {
+
+inline std::string GetIREECompilerPath() {
+  if (const char* path = std::getenv("XLA_OPENXLA_IREE_COMPILER_LIB")) {
+    return std::string(path);
+  }
+  return "libIREECompiler.so";
```

Comment:
It could be useful to add [verbose only?] logging here, for cases where the library can't be found. Hopefully less of an issue if you expect to be bundling the compiler as part of your build process and don't expect users to want to provide the path/library themselves.

Just as an example, here's some code we have in IREE: https://github.com/openxla/iree/blob/main/compiler/src/iree/compiler/Utils/ToolUtils.cpp

Reply:
Left it as is, because I'm not sure yet how exactly we'll be building it in open source, internally I expect we'll eventually get the static linking